src/main/java/org/elasticsearch/index/query/RangeQueryParser.java

```diff
@@ -123,11 +123,6 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         FieldMapper mapper = parseContext.fieldMapper(fieldName);
         if (mapper != null) {
             if (mapper instanceof DateFieldMapper) {
-                if ((from instanceof Number || to instanceof Number) && timeZone != null) {
```

Comment:
the problem here is, that a date like `2015121212` supplied as a number could be a date in the format `yyyyDDMMHH` which cannot be distinguished, so this check would trigger even though the number is valid and going to be parsed with a date that correctly owns a timezone. Not sure, when/where to postpone this check, need to think about it.